metabase: init at 0.27.2

schneefux opened the pull request:
###### Motivation for this change
Added the https://metabase.com "uberjar". Tested on NixOS, the only issue I found was a compatibility problem with MariaDB which has already been reported [upstream](https://github.com/metabase/metabase/issues/2581).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

schneefux committed:
metabase: init at 0.27.2

joachifm commented on pkgs/servers/metabase/default.nix:
```diff
@@ -0,0 +1,28 @@
+{ pkgs, stdenv, lib, fetchurl, makeWrapper, jre }:
```
The `pkgs` parameter looks unused.  Note also that `lib` can be (and is typically) accessed via `stdenv.lib`.

joachifm commented on pkgs/servers/metabase/default.nix:
```diff
@@ -0,0 +1,28 @@
+    sha256 = "1xsd6k362kajbf6sw0pb7zkd686i350fqqin858a5mmjlm5jkci7";
+  };
+
+  buildInputs = [ makeWrapper ];
```
`makeWrapper` can be treated as a `nativeBuildInputs`

joachifm commented on pkgs/servers/metabase/default.nix:
```diff
@@ -0,0 +1,28 @@
+
+  buildInputs = [ makeWrapper ];
+
+  phases = "installPhase";
```
Please do not explicitly list phases, instead disable those that are causing problems. While it may not matter to this package in particular, it is a common source of bugs.

joachifm commented on pkgs/servers/metabase/default.nix:
```diff
@@ -0,0 +1,28 @@
+  '';
+
+  meta = with lib; {
+    description = "Metabase is the easy, open source way for everyone in your company to ask questions and learn from data.";
```
Please simplify `Foo is a ...` to `A ...` and drop trailing punctuation.

joachifm reviewed (changes requested):
I think the `phases` and `description` issues must be addressed before merging, otherwise LGTM at a surface level.

schneefux committed:
metabase: 0.27.2 -> 0.28.1

schneefux commented:
> I think the phases and description issues must be addressed before merging, otherwise LGTM at a surface level.

I updated the package to 0.28.1, addressed your feedback and rebased on master.

joachifm commented on pkgs/servers/metabase/default.nix:
```diff
@@ -0,0 +1,28 @@
+{ pkgs, stdenv, fetchurl, makeWrapper, jre }:
```
`pkgs` seems to be unused

schneefux replied:
fixed

joachifm commented on pkgs/servers/metabase/default.nix:
```diff
@@ -0,0 +1,28 @@
+  installPhase = ''
+    mkdir -p $out/share/java
+    ln -s $jar $out/share/java/metabase.jar
+    makeWrapper ${jre}/bin/java $out/bin/metabase --add-flags "-jar $out/share/java/metabase.jar"
```
Not important, but I wonder if creating the dir/symlink is necessary; what if you just pass `-jar $jar`?
In that case, the whole expression reduces to, basically
```nix
writeScriptBin "metabase" ''
  #! /bin/sh
  exec ${jre}/bin/java -jar ${jar} "$@"
''
```
Just something to consider.

schneefux replied:
It is probably not necessary, I am only not very Nix experienced yet.
How does `writeScriptBin` work?
I tried
```nix
  installPhase = writeScriptBin "metabase" ''
    #!/bin/sh
    exec ${jre}/bin/java -jar ${jar} "$@"
  '';
```
But it fails with `/nix/store/…/*-metabase: Is a directory`. I grepped for usage examples but could only find packages where `writeScriptBin` is used in the `let` scope for dependencies.

joachifm replied:
`writeScriptBin` describes a derivation that given a name and text produces an output comprising an executable script of that name containing that text under a `bin/` directory.

In your case, `installPhase` is supposed to be a Nix string  (which is then exported to the builder); you're instead setting it to be a Nix derivation (evaluating to a store path). 

What I was suggesting was that the whole package expression would reduce, essentially, to `writeScriptBin`. That is, no `installPhase` or anything. I'm not suggesting you do literally that :)

In the context of your original expression, something like (untested)
```
installPhase = ''
  mkdir -p $out/bin
  cat >$out/bin/metabase <<EOF
  #! /bin/sh
  exec ${jre}/bin/java -jar ${jar} "$@"
  EOF
''
```
might work.

Anyway, this is not at all a blocker. Please just remove the redundant `pkgs` parameter and I'd be happy to merge as-is.

schneefux replied:
Thanks a lot for your explanation! :heart: I decided to stick with the verbose approach because it is more common.

> Not important, but I wonder if creating the dir/symlink is necessary; what if you just pass `-jar $jar`?

Fixed.

joachifm reviewed (approved):
A final nit.

schneefux committed:
metabase: refactor
